tsdb/engine/tsm1/compact.go

```diff
@@ -225,6 +225,10 @@ func (c *DefaultPlanner) FullyCompacted() (bool, string) {
 	} else if gens.hasTombstones() {
 		return false, "not fully compacted and not idle because of tombstones"
 	} else {
+		// Safety: check for first index so we don't accidentally do out of bounds access
+		if len(gens) == 1 && len(gens[0].files) > 1 && gens[0].size() < uint64(maxTSMFileSize) {
```

Comment:
Is the safety check `len(gens) == 1`, or `len(gens[0].files`?)

Reply:
len(gens) == 1, I want to ensure there is a 0th element I can index in to

Reply:
It looks like `gens[0].size()` would be fine if `gens[0].files` is empty. It feels like `len(gens[0].files) > 1` is there because we want to make sure there is at least one file because it's important for the planner logic. If so, commenting why that is there is important. On the first pass it was ambiguous to me if `len(gens[0].files > 1` as the bounds check referred to in the comment, or an important piece of logic.